src/main/resources/avro/referencemethods.avdl

```diff
@@ -325,4 +386,99 @@ GetSequenceBasesResponse getSequenceBases(
     /** Additional request parameters to restrict the query. */
     GetSequenceBasesRequest request) throws GAException;
 
+/****************  /joins/search  *******************/
+/**
+This request maps to the body of `POST /joins/search` as JSON. Specifies a
+number of filters, all of which must be satisfird by each result returned.
+*/
+record SearchJoinsRequest {
+  /**
+  If present, return only `Join`s which belong to this reference set.
+  */
+  union { null, string } referenceSetId = null;
+
+  /**
+  If present, return only `Join`s which belong to this variant set.
+  */
+  union { null, string } variantSetId = null;
+
+  // TODO: Specify a proper sort of filter object instead of this. We want to be
+  // able to support a list of sequences, but also a single sequence and a
+  // range.
+
```

Comment:
The semantics of this function is still obscure to me. Specifying a list of sequences seems fine, but the rest (starts, lengths, and strands)  seems very strange to me. What client operations are we trying to support here? Perhaps it would be better to break this into a number of different server API calls rather than trying to shoehorn all of this functionality into one call?